src/main/java/com/github/manolo8/darkbot/backpage/NovaManager.java

```diff
@@ -0,0 +1,205 @@
+import com.github.manolo8.darkbot.utils.http.Method;
+import com.google.gson.Gson;
+import com.google.gson.JsonObject;
+
+import java.io.IOException;
+
+public class NovaManager {
+    private final BackpageManager backpageManager;
+    private final NovaData data;
+    private final Gson g;
+    private long lastNovaUpdate;
+
+    public NovaManager(BackpageManager backpageManager) {
+        this.backpageManager = backpageManager;
+        this.data = new NovaData();
+        this.g = new Gson();
+    }
+
+    public boolean update(long expiryTime) {
+        try {
+            if (System.currentTimeMillis() <= lastNovaUpdate + expiryTime) return false;
+
+            data.getRosterList().forEach((k, v) -> v.setForRemoval(true));
+
+            updateActiveCaptain(0);
+            updateResource();
+            updateRosterList();
+
+            data.getRosterList().values().removeIf(Agent::getForRemoval);
+            lastNovaUpdate = System.currentTimeMillis();
+
+            return true;
+        } catch (Exception e) {
+            e.printStackTrace();
+        }
+        return false;
+    }
+
+    public NovaData getData() {
+        return this.data;
+    }
+
+
+    /**
+     * This method is used to unequip, equip and get current captain
+     * @param id
+     * @throws IOException
+     */
+    public boolean updateActiveCaptain(int id) throws IOException {
+        String response = backpageManager.getConnection("ajax/captain.php", Method.POST)
+                .setRawParam("command", "updateActiveCaptain")
+                .setRawParam("captainId", id == 0 ? "" : id)
+                .getContent();
+
+        JsonObject jsonObj = g.fromJson(response, JsonObject.class); //Converts the json string to JsonElement without POJO
+        if (jsonObj.get("result").getAsString().equalsIgnoreCase("OK")) {
+            this.data.setActiveCaptainId(jsonObj.get("activeCaptainId").getAsString().isEmpty() ? 0 : jsonObj.get("activeCaptainId").getAsInt());
+            return true;
+        } else {
+            System.out.println("NovaManager: " + response);
+        }
+        return false;
+    }
+
+    /**
+     * Method to get resources count
+     * @throws IOException
+     */
+
+    private void updateResource() throws IOException {
+        String response = backpageManager.getConnection("ajax/captain.php", Method.POST)
+                .setRawParam("command", "getResources")
+                .getContent();
+
+        JsonObject jsonObj = g.fromJson(response, JsonObject.class); //Converts the json string to JsonElement without POJO
+        if (jsonObj.get("result").getAsString().equalsIgnoreCase("OK")) {
+            this.data.setResourceAmount(jsonObj.get("item").getAsJsonObject().get("amount").getAsInt());
+        } else {
+            System.out.println("NovaManager: " + response);
+        }
+    }
+
+    /**
+     * Method to get list of Nova Agent/Captain
+     * @throws IOException
+     */
```

Comment:
these comments are pretty meaningless with such vague descriptions
also, it's straight up wrong as this doesn't *get a list*, it doesn't return anything, it just updates them